nat/traversal/pinger.go

```diff
@@ -234,32 +209,32 @@ func (p *Pinger) BindServicePort(key string, port int) {
 	p.natProxy.registerServicePort(key, port)
 }
 
-func (p *Pinger) pingReceiver(conn *net.UDPConn, stop <-chan struct{}) error {
+func (p *Pinger) pingReceiver(conn *net.UDPConn, stop <-chan struct{}) (*net.UDPAddr, error) {
 	timeout := time.After(p.pingConfig.Timeout)
 	buf := make([]byte, bufferLen)
 
 	for {
 		select {
 		case <-timeout:
-			return errNATPunchAttemptTimedOut
+			return nil, errNATPunchAttemptTimedOut
 		case <-stop:
-			return errNATPunchAttemptStopped
+			return nil, errNATPunchAttemptStopped
 		case <-p.stop:
-			return errNATPunchAttemptStopped
+			return nil, errNATPunchAttemptStopped
 		default:
 			// Add read deadline to prevent possible conn.Read hang when remote peer doesn't send ping ack.
 			conn.SetReadDeadline(time.Now().Add(p.pingConfig.Timeout * 2))
-			n, err := conn.Read(buf)
+			n, raddr, err := conn.ReadFromUDP(buf)
 			// Reset read deadline.
 			conn.SetReadDeadline(time.Time{})
 
 			if err != nil || n == 0 {
-				log.Error().Err(err).Msgf("Failed to read remote peer: %s - attempting to continue", conn.RemoteAddr().String())
+				log.Error().Err(err).Msgf("Failed to read remote peer: %s - attempting to continue", raddr)
```

Comment:
It don't think it's ERR level for each parallel ping result.

Reply:
Changed it to debug.